onnxruntime/test/providers/cpu/tensor/split_op_test.cc

```diff
@@ -613,5 +616,25 @@ TEST(SplitOperatorTest, Uint8NegativeAxis) {
   RunTest<uint8_t>(axis, {}, input, outputs, false);
 }
 
+TEST(SplitOperatorTest, MissingOptionalInputAdded) {
```

Comment:
This probably is out the scope of this change. But just curious, based on the spec, if the optional `split` input is mssing, the tensor is split to equal sized parts, but what if the dimension in that axis cannot be equally divided into 2? Will ORT error out in this case?

Reply:
Yes, this is enforced on line 74 in PrepareForCompute() (and yes a failure status will be returned back to the user)

  if (split_sizes.empty()) {
    // equal split based on number of outputs
    if (split_dim_size % static_cast<size_t>(num_outputs) != 0) {
      return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Input cannot be split evenly on selected axis. Input shape=", input_shape,
                             " Axis=", axis_, " NumOutputs=", num_outputs);
    }